akka-projection-core/src/main/scala/akka/projection/internal/ProjectionSettings.scala

```diff
@@ -50,7 +50,12 @@ private[projection] object ProjectionSettings {
       val maxRestarts = restartBackoffConfig.getInt("max-restarts")
       if (maxRestarts >= 0) RestartSettings(minBackoff, maxBackoff, randomFactor)
       else RestartSettings(minBackoff, maxBackoff, randomFactor).withMaxRestarts(maxRestarts, minBackoff)
-    }
+    }.withLogSettings(
+      RestartSettings.LogSettings.defaultSettings
+        .withLogLevel(LogLevels.Warning)
+        // Once we have retried many times, it could still be a transient failure but is
+        // more likely to be a permanent problem, so increase verbosity/include full stack trace
+        .withVerboseLogsAfter(5))
```

Comment:
Not sure if this should perhaps be from config and tuneable or if just having a hardcoded sensible default is enough. With the default settings this means that it starts printing a full stack trace after retrying and backing off 5 times ( ~1 minute and 20 seconds.)

Reply:
We could also just explicitly define it in gRPC replication, and recommend/show setting it for direct gRPC usage instead of defaulting here.